tests/flytekit/unit/extras/pydantic/test_pydantic_transformer.py

```diff
@@ -0,0 +1,775 @@
+import os
+import tempfile
+from dataclasses import field
+from enum import Enum
+from typing import Dict, List, Optional, Union
+from pydantic import BaseModel, Field
+from unittest.mock import patch
+import pytest
+from google.protobuf import json_format as _json_format
+from google.protobuf import struct_pb2 as _struct
+
+from flytekit import task, workflow
+from flytekit.core.context_manager import FlyteContextManager
+from flytekit.core.type_engine import TypeEngine
+from flytekit.models.literals import Literal, Scalar
+from flytekit.types.directory import FlyteDirectory
+from flytekit.types.file import FlyteFile
+from flytekit.types.schema import FlyteSchema
+from flytekit.types.structured import StructuredDataset
+
+
+class Status(Enum):
+    PENDING = "pending"
+    APPROVED = "approved"
+    REJECTED = "rejected"
+
+
+@pytest.fixture
+def local_dummy_file():
+    fd, path = tempfile.mkstemp()
+    try:
+        with os.fdopen(fd, "w") as tmp:
+            tmp.write("Hello FlyteFile")
+        yield path
+    finally:
+        os.remove(path)
+
+
+@pytest.fixture
+def local_dummy_directory():
+    temp_dir = tempfile.TemporaryDirectory()
+    try:
+        with open(os.path.join(temp_dir.name, "file"), "w") as tmp:
+            tmp.write("Hello FlyteDirectory")
+        yield temp_dir.name
+    finally:
+        temp_dir.cleanup()
+
+
+def test_flytetypes_in_pydantic_basemodel_wf(local_dummy_file, local_dummy_directory):
+    class InnerBM(BaseModel):
+        flytefile: FlyteFile = field(default_factory=lambda: FlyteFile(local_dummy_file))
+        flytedir: FlyteDirectory = field(default_factory=lambda: FlyteDirectory(local_dummy_directory))
+
+    class BM(BaseModel):
```

Comment:
maybe in a different file, but can we add pydantic models that contain `dataclass` and also `dataclass`es that contain Pydantic models?  I know some people have been asking for that, be good to have some tests for it.

Thank you!

Reply:
> pydantic models that contain dataclass
this is supported

> dataclasses that contain Pydantic models
this is never supported, if people really want it, we can do it in a seperate PR

Reply:
I'll add unit tests today, thank you so much